### DISCUSSION TITLE
Similar project using FreeRTOS

### DISCUSSION THREAD

**Discussion by giovani-luigi** [MaJerle/lwshell#2] [Show and tell]
Mr. MaJerle,

I am an embedded system enthusiast too, and about a year ago I've made a similar project (just for fun) to learn FreeRTOS on the STM32F746 demo board. Obviously, it was not supposed to be something to re-use in large projects... perhaps some small things I would do myself. I've employed FreeRTOS because (apart from me learning it) it seemed that most of the time where I would have a command-line shell, I would probably be using it to debug/control/service some running system like a real-time controller, which would require multi-threading. That dependency obviously reduces flexibility and re-use cases. But on the other hand, my personal target was high-end MCUs, where having a simple and free RTOS seemed a reasonable thing.

Anyway, I am just commenting, if you want to have a look at it. It's a public repo in my account here.
I know you work for STM32 and you are probably able to implement it much better than an STM32 beginner like me.
Maybe I can become a contributor to your project, once I figure out some free time kkkk
I took a break from MCUs code for a while, but maybe a good thing I come back to focus on these projects.

[Link to my code](https://github.com/giovani-luigi/uConsole)

Anyway... Congrats on your project!

Giovani Luigi

**Comment by MaJerle** [comment]
I see you did a good work there, highly focused on FreeRTOS. Here I go different direction, I let user to use it anywhere and use any input/output functionality :)

Glad to hear it from you.
